src/main/java/com/endava/endabank/security/utils/JwtManage.java

```diff
@@ -5,45 +5,48 @@
 import com.auth0.jwt.algorithms.Algorithm;
 import com.auth0.jwt.interfaces.Claim;
 import com.auth0.jwt.interfaces.DecodedJWT;
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.security.core.userdetails.UserDetails;
-import org.springframework.stereotype.Component;
+import com.endava.endabank.constants.Strings;
+import com.endava.endabank.exceptions.customexceptions.BadDataException;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
 
-import java.nio.charset.StandardCharsets;
 import java.util.Date;
 import java.util.HashMap;
 import java.util.Map;
 
-@Component
-public class JwtManage {
-    @Value("${jwt.secret}")
-    private String secret;
 
-    public String generateToken(UserDetails userDetails) {
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class JwtManage {
+
+    public static String generateToken(Integer id, String username, String secret) throws BadDataException {
         Map<String, Object> claims = new HashMap<>();
-        return doGenerateToken(claims, userDetails.getUsername());
+        if (id == null || username == null || "".equals(username)
```

Comment:
Hi,

If you use a method to have this kind of block of code like this to check something and throw an exception, you can test it easier and keep this method more readable.